cli/cmd/proxy/read/filters.go

```diff
@@ -0,0 +1,97 @@
+package read
+
+import (
+	"strconv"
+	"strings"
+)
+
+// FilterClusters takes a slice of clusters along with parameters for filtering
+// those clusters.
+//
+// - `fqdn` filters clusters to only those with fully qualified domain names
+//   which contain the given value.
+// - `address` filters clusters to only those with endpoint addresses which
+//   contain the given value.
+// - `port` filters clusters to only those with endpoint addresses with ports
+//   that match the given value. If -1 is passed, no filtering will occur.
+//
+// The filters are applied in combination such that a cluster must adhere to
+// all of the filtering values which are passed in.
+func FilterClusters(clusters []Cluster, fqdn, address string, port int) []Cluster {
+	// No filtering no-op.
+	if fqdn == "" && address == "" && port == -1 {
+		return clusters
+	}
+
+	portStr := ":" + strconv.Itoa(port)
```

Comment:
Not here, but do you handle if someone doesn't pass an integer as a port number?  ie, they put in `:8000` ?